### DISCUSSION TITLE
localized routing with translated paths ?

### DISCUSSION THREAD

**Discussion by philippeauriach** [vercel/next.js#26469] [Ideas]
### Describe the feature you'd like to request

I would like to have translated path to my pages.

For example those two pages are the "same" except for the language used:
- mynextsite.com/about
- mynextsite.com/fr/a-propos

But those urls should NOT exists:
- mynextsite.com/a-propos (could redirect to /fr/a-propos or 404)
- mynextsite.com/fr/about (could redirect to /about or 404)

### Describe the solution you'd like

Maybe we could have a GetPageSlug method exported from the page, that will return the list of slugs handled by this page (in addition to the name of the file exporting the page). Next would make sure all each of them is unique in it's own language.

Example
```javascript
// src/pages/about.tsx

function About() {
  return <>
    <AboutComponent />
  </>
}

export const getPageSlugs = () => {
  return {fr: 'a-propos'}
}

export default About
```

### Describe alternatives you've considered

The alternative I use now is to have a "handle all" page in which I check the path against my translations, and then return the correct component. Which loses a bit of the purpose of next :D 

![Capture d’écran 2021-06-22 à 10 04 52](https://user-images.githubusercontent.com/920265/122887530-4a6e2700-d341-11eb-804f-4b000e2d495b.png)

**Comment by moonmeister** [comment]
Wanted the same thing. My thought was allowing a function in the i18n section of the config that resolved the other languages but a page export would work and be more consistent with existing apis.

**Comment by thomasverleye** [comment]
Probably too late for that but have you read into rewrites?
https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewrites-with-i18n-support

**Reply by philippeauriach** [reply]
That would work ! A bit weird to have it separated from the page file, but it would definitely work, if I generated a redirect for each locale. Thanks ! A bit late for this project but definitely useful for the next one !

**Reply by thomasverleye** [reply]
It's still not ideal indeed.
I really hope there will be a better solution for this in the near future.
Especially dynamic routes are hard to figure out which slug is connected with a different locale (for Language Switching etc).
Discussions like these hopefully bump up the feature requests 🤞 

Anyway, Hopefully this helps for you future projects, best of luck!
